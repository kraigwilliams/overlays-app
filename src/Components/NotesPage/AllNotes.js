import React,{Component} from 'react';
import config from '../../config'
import TokenService from '../../services/token-service'

class AllNotes extends Component{
    
    state={
        notes:[{
            id:"4",
            note_title:"stuff",
            note_contents:"this is really great"
        }]
    }

    componentDidMount(){
        const topicId= this.props.match.params.topicId
        console.log("match object",this.props.match, "state",this.state.notes)

        fetch(`${config.API_ENDPOINT}/notes/${topicId}`,{
            headers:{
                authorization:"bearer " + TokenService.getAuthToken()
              }
        })
        .then(res=> res.json())
        .then(data=>{
            console.log("this is data", data)
            this.setState({notes:data})
                      
        }
            )

    }
render(){

    return(

<>
{this.state.notes.map(note=>{
  return  <>
<h3 >{note.note_title}</h3>
<p>{note.note_contents}</p>
</>
})}



</>

    )
}




}

export default AllNotes