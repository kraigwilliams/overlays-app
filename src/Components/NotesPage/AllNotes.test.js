import React from 'react';
//import { render } from '@testing-library/react';
import AllNotes from './AllNotes'
import {BrowserRouter} from 'react-router-dom'
import ReactDOM from  'react-dom';
import renderer from 'react-test-renderer';

describe('<AllNotes/>',()=>{
  it ('is able to render',()=>{
    const div = document.createElement('div')
    const match={
        params:{
        topicName:'great'
        }
    };

    
    ReactDOM.render(<BrowserRouter><AllNotes match={match} /></BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders the UI as expected', () => {
    const match={
            params:{
            topicName:'great'
            }
        };
    
    const tree = renderer
      .create(<BrowserRouter><AllNotes match={match} /></BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });

  
})