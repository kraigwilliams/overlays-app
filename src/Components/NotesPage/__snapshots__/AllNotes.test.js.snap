// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AllNotes/> renders the UI as expected 1`] = `
Array [
  <h2>
    
  </h2>,
  <p>
    <a
      href="/new-note"
      onClick={[Function]}
    >
      Add New Note
    </a>
  </p>,
]
`;
